### ISSUE TITLE
Add support for Unit structs/enums ?

### ISSUE THREAD

**Issue by chifflier** [rust-bakery/nom-derive#6]
Followup from #5 
If `Selector` attribute is provided (and maybe `Parse`), it may be possible to parse a Unit struct/enum. Something like:
```rust
#[nom(Selector="u8")]
pub enum U {
  #[nom(Selector="0")]
  Foo,
  #[nom(Selector="_")]
  Bar
}
```

Remark: the annotations/structure declaration ratio is not very efficient

**Comment by chifflier** [comment]
Additional note: the example above is equivalent to a `From` implementations, with extra steps ...

**Comment by chifflier** [comment]
Doing some cleanup, so closing:
- fieldless enums (example from this issue) are supported using selector
- fieldless enums with a `#[repr(<type>)]` are implemented in f68928426a32490f903465557b0e014b0909a4c4
- enums with values in variants are already supported (named and unnamed)
